xdsl/dialects/riscv.py

```diff
@@ -331,6 +334,73 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        operands = cls.parse_operands(parser)
+        successors = cls.parse_successors(parser)
+        regions = cls.parse_regions(parser)
+        attributes = cls.parse_attributes(parser)
+        parser.parse_punctuation(":")
+        result_types = cls.parse_function_type(parser).outputs.data
+        return cls.create(
+            operands=operands,
+            result_types=result_types,
+            attributes=attributes,
+            successors=successors,
+            regions=regions,
+        )
+
+    @classmethod
+    def parse_operands(cls, parser: Parser) -> Sequence[SSAValue]:
+        return parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, parser.parse_operand
+        )
+
+    @classmethod
+    def parse_successors(cls, parser: Parser) -> Sequence[Block]:
+        return parser.parse_optional_successors() or []
+
+    @classmethod
+    def parse_function_type(cls, parser: Parser) -> FunctionType:
+        return parser.parse_function_type()
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        return parser.parse_optional_attr_dict()
```

Comment:
I would remove these two and replace them with calls directly to the method they wrap. I don't see a point in having them otherwise.

Reply:
I agree.

Reply:
This is left for subsequent PR which will / may print custom function type: `(zero, t0) -> (t1)` instead of `(!riscv.reg<zero>, !riscv.reg<t0>) -> (!riscv.reg<t1>)`.

Reply:
Similarly for attributes. Current PR tries to minimize the diff in `tests/filecheck`.

Reply:
We may need these later, but for now I would just call `parser.parse_optional_attr_dict()` directly and do not yet introduce these `parse_attributes()` functions. They are not yet needed, and introducing abstractions without use cases is typically difficult to review, if the point of this PR is to not yet open too many discussions at once.

Reply:
> I would remove these two and replace them with calls directly to the method they wrap. I don't see a point in having them otherwise.

Can you follow @AntonLydike's advice here? It is hard to establish abstractions without justifications, and the point of this PR is to limit the scope of discussions to a minimum. So we do not want to pull in other questions to provide evidence why these abstractions are needed. I feel we can introduce them iff they are needed (and I am not even sure we will need them).

Reply:
Removed the function, will be exposed in next PR.